implants/eldritch/src/pivot/ncat_impl.rs

```diff
@@ -0,0 +1,177 @@
+
+use std::net::Ipv4Addr;
+
+use anyhow::Result;
+use tokio::io::{AsyncWriteExt, AsyncReadExt};
+use tokio::net::{TcpStream, UdpSocket};
+
+// Since we cannot go from async (test) -> sync (ncat) `block_on` -> async (handle_ncat) without getting an error "cannot create runtime in current runtime since current thread is calling async code."
+async fn handle_ncat(address: String, port: i32, data: String, protocol: String) -> Result<String> {
+    // If the response is longer than 4096 bytes it will be  truncated.
+    let mut response_buffer = [0; 4096];
```

Comment:
Not sure if there's a better way to receive more bytes.
If not should we add a truncated indicator? Eg. "I got. lots of bytes of data ...trunc...

Reply:
Can we use `read_exact` either to read enough header to determine the packet size, or alternatively read_exact in a loop?

https://docs.rs/tokio/0.1.22/tokio/io/fn.read_exact.html

Reply:
Good suggestion.
Do you know of a good way to read an undefined number of bytes? I think this still uses a fixed length buffer.